test/integration/network_ips_test.go

```diff
@@ -132,6 +192,184 @@ func TestIPAddress_Update(t *testing.T) {
+		Reserved: true,
+	}
+	updatedIP, err := client.UpdateIPAddress(context.Background(), ephemeralIP, updateOpts)
+	if err != nil {
+		t.Fatalf("Failed to convert ephemeral IP to reserved: %v", err)
+	}
+	if !updatedIP.Reserved {
+		t.Errorf("Expected IP to be reserved, but it's not")
+	}
+
+	// Scenario 2: Convert reserved IP to reserved IP (no-op)
+
+	reservedIP, err := createReservedIP()
+	if err != nil {
+		t.Fatalf("Failed to create reserved IP: %v", err)
+	}
+	defer client.DeleteReservedIPAddress(context.Background(), reservedIP)
+
+	updateOpts = IPAddressUpdateOptions{
+		Reserved: true,
+	}
+	updatedIP, err = client.UpdateIPAddress(context.Background(), reservedIP, updateOpts)
+	if err != nil {
+		t.Fatalf("Failed to update reserved IP: %v", err)
+	}
+	if !updatedIP.Reserved {
+		t.Errorf("Expected IP to remain reserved, but it's not")
+	}
+
+	// Scenario 3: Convert reserved to ephemeral
+
+	ephemeralIP = instance.IPv4[0].String()
+	updateOpts = IPAddressUpdateOptions{
+		Reserved: false,
+	}
+	updatedIP, err = client.UpdateIPAddress(context.Background(), ephemeralIP, updateOpts)
+	if err != nil {
+		t.Fatalf("Failed to update ephemeral IP: %v", err)
+	}
+	if updatedIP.Reserved {
+		t.Errorf("Expected IP to remain ephemeral, but it's reserved")
+	}
+
+	// Scenario 4: Convert assigned reserved IP to ephemeral
+	reservedIP, err = createReservedIP()
+	if err != nil {
+		t.Fatalf("Failed to create reserved IP: %v", err)
+	}
+	defer client.DeleteReservedIPAddress(context.Background(), reservedIP)
+
+	// Assign the reserved IP to the instance
+	assignOpts := LinodesAssignIPsOptions{
+		Region: instance.Region,
+		Assignments: []LinodeIPAssignment{
+			{
+				Address:  reservedIP,
+				LinodeID: instance.ID,
+			},
+		},
+	}
+	err = client.InstancesAssignIPs(context.Background(), assignOpts)
+	if err != nil {
+		t.Fatalf("Failed to assign reserved IP: %v", err)
+	}
+
+	updateOpts = IPAddressUpdateOptions{
+		Reserved: false,
+	}
+	updatedIP, err = client.UpdateIPAddress(context.Background(), reservedIP, updateOpts)
+	if err != nil {
+		t.Fatalf("Failed to convert assigned reserved IP to ephemeral: %v", err)
+	}
+	if updatedIP.Reserved {
+		t.Errorf("Expected IP to be converted to ephemeral, but it's still reserved")
+	}
+
+	// Sceanrio 5: Cannot set RDNS for unassigned reserved IP
+
+	unassignedResIP, unassignedResIpErr := createReservedIP()
+	if unassignedResIpErr != nil {
+		t.Fatalf("Failed to create reserved IP: %v", unassignedResIpErr)
+	}
+
+	updateOpts = IPAddressUpdateOptions{
+		Reserved: true,
+		RDNS:     String("sample rdns"),
+	}
+	_, err = client.UpdateIPAddress(context.Background(), unassignedResIP, updateOpts)
+	if err == nil {
+		t.Fatalf("Expected error when setting RDNS for unassigned reserved IP, but got none")
+	}
+
+	// Sceanrio 6: Try to reserve an IP at MAX reserve IP limit
+
+	reservedIP, err = createReservedIP()
+	if err == nil {
+		t.Fatalf("Expected error indicating MAX IP Reservation limit has been reached, got nil")
```

Comment:
We typically try to design tests to be account/limit-agnostic so I think we can remove this case

Reply:
Got it! I will go ahead and remove this test.